apps/prairielearn/src/components/AssessmentBadge.html.ts

```diff
@@ -0,0 +1,38 @@
+import { html } from '@prairielearn/html';
+
+export function AssessmentBadge({
+  assessment,
+  hideLink = false,
+  urlPrefix,
+  plainUrlPrefix,
+  course_instance_id,
+}: {
+  assessment: { assessment_id: string; color: string; label: string };
+  hideLink?: boolean;
+} & ( // If urlPrefix is not provided, then plainUrlPrefix and course_instance_id must be provided and the appropriate URL prefix will be constructed
+  | {
+      urlPrefix: string;
+      plainUrlPrefix?: unknown;
+      course_instance_id?: unknown;
```

Comment:
Nit, for symmetry with the other type (untested):

```suggestion
      plainUrlPrefix?: undefined;
      course_instance_id?: undefined;
```

Reply:
I used unknown here to trigger an error if these values end up being used somehow in this context. I'm ok with undefined as well. Change coming up.